core/src/channel/mod.rs

```diff
@@ -249,6 +267,37 @@ impl VoiceChannelData {
                         self.process_control_event(ControlEvent::PitchBendSensitivity(sensitivity))
                     }
                 }
+                0x07 => {
+                    // Volume
+                    let vol: f32 = value as f32 / 128.0;
+                    self.control_event_data
+                    .borrow_mut()
+                    .volume = vol
+                }
+                0x0A => {
+                    // Pan
+                    let pan: f32 = value as f32 / 128.0;
+                    self.control_event_data
+                    .borrow_mut()
+                    .pan = pan
+                }
+                0x40 => {
+                    // Damper / Sustain
+                    let damper = match value {
+                        0..=63 => false,
+                        64..=127 => true,
+                        _ => false,
+                    };
+
+                    for key in self.key_voices.iter() {
+                        key.data.borrow().voices.damper = damper;
+                        key.data.borrow().voices.release_next_voice();
```

Comment:
Why do we need to release the next voice here?

Reply:
I thought about adding a comment there, don't know why I ended up not doing it.

Basically I am calling this function every time the damper event changes, in case it changes to false and the "held" voices need to be released.
Otherwise, if the status of the damper changes, the voices will only be released on the next Note Off event.

Maybe it would be better to rename the function to something like `check_to_release_voices`, or to modify its comment, since it was basically repurposed.